net: l2: ppp: ip{,v6}cp: remove network address in ip{,v6}cp_down

mniestroj opened the pull request:
Make sure IPv{4,6} address is removed from network interface in IP{,V6}CP
protocol down handler. This makes sure that application can receive
high-level notification about missing network connection.

mniestroj committed:
net: l2: ppp: ipcp: remove network address in ipcp_down

Make sure IPv4 address is removed from network interface in IPCP
protocol down handler. This makes sure that application can receive
high-level notification about missing network connection.

Signed-off-by: Marcin Niestroj <m.niestroj@grinn-global.com>

jukkar reviewed (approved):
Good point. It looks like similar thing is needed for IPv6, would you be able to add a commit about that to this PR?

mniestroj committed:
net: l2: ppp: ipv6cp: remove network address in ipv6cp_down

Make sure IPv6 address is removed from network interface in IPV6CP
protocol down handler. This makes sure that application can receive
high-level notification about missing network connection.

Signed-off-by: Marcin Niestroj <m.niestroj@grinn-global.com>

mniestroj commented:
@jukkar Added new commit for ipv6cp as requested.
